### PR TITLE
Optimizing allocations using Stream::size_hint

### PR DESCRIPTION
**Pull request by k-nasa**
close: https://github.com/async-rs/async-std/issues/128
Is this the intention of this issue?

### CODE REVIEW

File: src/collections/linked_list/extend.rs [hunk 1]
```diff
@@ -10,9 +10,6 @@ impl<T> Extend<T> for LinkedList<T> {
         stream: S,
     ) -> Pin<Box<dyn Future<Output = ()> + 'a>> {
         let stream = stream.into_stream();
-        //TODO: Add this back in when size_hint is added to Stream/StreamExt
-        //let (lower_bound, _) = stream.size_hint();
-        //self.reserve(lower_bound);
```
Comment (yoshuawuyts) [inline src/collections/linked_list/extend.rs]: Did you intend to add the size hint?

Comment (yoshuawuyts) [inline src/collections/linked_list/extend.rs]: Did you intend to add the size hint?

Comment (k-nasa) [reply src/collections/linked_list/extend.rs]: sorry! I forgot to ask a question.
A LinkedList should consist of a chain of pointers. So, isn't it difficult to make a size or change the size?

Comment (yoshuawuyts) [reply src/collections/linked_list/extend.rs]: @k-nasa oh yeah, that's accurate! I see what you mean; I agree this doesn't make sense.

**Pull request merged**